src/options/output.rs

```diff
@@ -0,0 +1,203 @@
+        let output = match format_type {
+            FormatType::Plain => self.get_stdout(),
+            FormatType::Json => Ok(Some(JsonOutput::from(self.clone()).to_string())),
+        };
+
+        // Print the RoverOutput to file or stdout.
+        if let Ok(Some(result)) = output {
+            match output_type {
+                FinalOutputType::File(path) => {
+                    let success_heading = Style::Heading.paint("The output was printed to ");
+                    let path_text = Style::Path.paint(&path);
+                    Fs::write_file(&path, result)?;
+                    stderrln!("{} {}", success_heading, path_text)?;
+                }
+                FinalOutputType::Stdout => {
+                    // Call the appropriate method based on the variant of RoverOutput.
+                    let descriptor = match &self {
+                        RoverOutput::FetchResponse(fetch_response) => {
+                            match fetch_response.sdl.r#type {
+                                SdlType::Graph | SdlType::Subgraph { .. } => "SDL",
+                                SdlType::Supergraph => "Supergraph SDL",
+                            }
+                        }
+                        RoverOutput::CoreSchema(_) => "CoreSchema",
+                        RoverOutput::CompositionResult(_) => "CoreSchema",
+                        RoverOutput::TemplateUseSuccess { .. } => "Project generated",
+                        RoverOutput::CheckResponse(_) => "Check Result",
+                        RoverOutput::AsyncCheckResponse(_) => "Check Started",
+                        RoverOutput::Profiles(_) => "Profiles",
+                        RoverOutput::Introspection(_) => "Introspection Response",
+                        RoverOutput::ReadmeFetchResponse { .. } => "Readme",
+                        RoverOutput::GraphPublishResponse { .. } => "Schema Hash",
+                        _ => return Ok(()),
+                    };
+                    if let RoverOutput::GraphPublishResponse { .. } = self {
+                        self.print_one_line_descriptor(descriptor)?;
+                    } else {
+                        self.print_descriptor(descriptor)?;
+                    }
+
+                    stdoutln!("{}", &result)?;
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl RoverOutputTrait for RoverError {
+    fn write_or_print(self, format_type: FormatType, _: FinalOutputType) -> RoverResult<()> {
+        match format_type {
+            FormatType::Plain => self.print(),
+            FormatType::Json => JsonOutput::from(self).print(),
+        }?;
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Parser, Serialize)]
+pub struct OutputStrategy {
+    /// Specify Rover's format type
+    #[arg(long = "format", global = true)]
+    format_type: Option<FormatType>,
+
+    /// Specify a file to write Rover's output to
+    #[arg(long = "output", global = true)]
```

Comment:
in line with the CLI guideline link in the PR descriptor, let's add a `-o` shorthand for this. you can do like this:

```suggestion
    #[arg(long = "output", short = 'o', global = true)]
```